Cauldron/Assets/Scripts/GameModes/PrestigeGameMode.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using CauldronCodebase.GameStates;
+using UnityEngine;
+using Zenject;
+
+namespace CauldronCodebase
+{
+    public class PrestigeGameMode : GameModeBase
+    {
+        [SerializeField] private int targetDays = 25;
+        protected string achievIdents;
+        
+        protected GameStateMachine gameStates;
+        protected GameDataHandler gameData;
+        private IAchievementManager achievement;
+
+        [Inject]
+        public void Construct(GameStateMachine gameStates, GameDataHandler gameData, IAchievementManager achievement)
+        {
+            this.gameStates = gameStates;
+            this.achievement = achievement;
+            this.gameData = gameData;
+        }
+        
+        public override void Apply()
```

Comment:
Here we can make an abstract OnApply method and call it inside Apply() after the subscription.